src/app/api/services/edit-page/edit-page.service.ts

```diff
@@ -3,22 +3,151 @@ import { Injectable } from '@angular/core';
+        return this.get(url, 'PREVIEW_MODE');
+    }
+
+    /**
+     * Get the page HTML in live mode
+     *
+     * @param {string} url
+     * @returns {Observable<DotRenderedPage>}
+     * @memberof EditPageService
+     */
+    getLive(url: string): Observable<DotRenderedPage> {
+        return this.get(url, 'LIVE_MODE');
+    }
+
+    /**
+     * Lock a content asset
+     *
+     * @param {string} inode
+     * @returns {Observable<any>}
+     * @memberof PageViewService
+     */
+    lock(inode: string): Observable<any> {
+        return this.coreWebService.requestView({
+            method: RequestMethod.Put,
+            url: `content/lock/inode/${inode}`
+        });
+    }
+
+    /**
+     * Unlock a content asset
+     *
+     * @param {string} inode
+     * @returns {Observable<any>}
+     * @memberof PageViewService
+     */
+    unlock(inode: string): Observable<any> {
+        return this.coreWebService.requestView({
+            method: RequestMethod.Put,
+            url: `content/unlock/inode/${inode}`
+        });
+    }
+
+    /**
+     * Set the page state
+     *
+     * @param {DotRenderedPage} page
+     * @param {DotEditPageState} state
+     * @returns {Observable<any>}
+     * @memberof EditPageService
+     */
+    setPageState(page: DotRenderedPage, state: DotEditPageState): Observable<DotRenderedPageState> {
+
+        const methods = [];
+        let lockUnlock;
+        let pageMode;
+
+        // When lock prop is not set is null
+        if (state.lock === true) {
+            lockUnlock = this.lock(page.liveInode)
+                .pluck('bodyJsonObject')
+                .pluck('message');
+        } else if (state.lock === false) {
+            lockUnlock = this.unlock(page.liveInode)
+                .pluck('bodyJsonObject')
+                .pluck('message');
+
+        }
+
+        if (state.mode === PageMode.PREVIEW) {
+            pageMode = this.getPreview(page.pageUri);
+        } else if (state.mode === PageMode.EDIT) {
+            pageMode = this.getEdit(page.pageUri);
+        } else if (state.mode === PageMode.LIVE) {
+            pageMode = this.getPreview(page.pageUri);
+        }
+
+        if (lockUnlock && pageMode) {
+            return lockUnlock.mergeMap((lockState: string) => {
+                return pageMode.map((dotRenderedPage: DotRenderedPage) => {
+                    return {
+                        dotRenderedPage: dotRenderedPage,
+                        lockState: lockState
+                     };
+                });
+            });
+        } else if (lockUnlock) {
+            return lockUnlock.map((lockState: string) => {
+                return {
+                    lockState: lockState
+                };
+            });
+        } else if (pageMode) {
+            return pageMode.map((dotRenderedPage: DotRenderedPage) => {
+                return {
+                    dotRenderedPage: dotRenderedPage
+                };
+            });
+        }
```

Comment:
maybe we can clean up the code allways setting something to pageMode and lockUnLock, we can do something like:

if (state.lock === true) {
    ....
} else if (state.lock === false) {
   ....
} else {
    lockUnlock = Observable.of({})
}

and the same for pageMode